### PR TITLE
Docs: various improvements.

### PR DESCRIPTION
**Pull request by concretevitamin**
Docs
- added a few sections based on frequently asked questions

Minor change in Resources display
- This can be triggered by running the following
```yaml
  resources:
    # SkyPilot will perform cost optimization among the specified regions.
    any_of:
      # AWS:
      - region: eu-central-1
      - region: eu-west-1
      - region: eu-west-2
      - region: eu-west-3
      - region: eu-north-1
      # GCP:
      - region: europe-central2
      - region: europe-north1
      - region: europe-southwest1
      - region: europe-west1
      - region: europe-west10
      - region: europe-west12
      - region: europe-west2
      - region: europe-west3
      - region: europe-west4
      - region: europe-west6
      - region: europe-west8
      - region: europe-west9
```
```console
» sky launch test-gdpr.yaml --gpus T4
Task from YAML spec: test-gdpr.yaml
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-southwest1) on GCP.
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-west6) on GCP.
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-west10) on GCP.
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-north1) on GCP.
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-west8) on GCP.
I 11-28 20:52:25 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-west12) on GCP.
I 11-28 20:52:26 optimizer.py:1223] No resource satisfying GCP({'T4': 1}, region=europe-west9) on GCP.
...
```

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered docs
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/docker-containers.rst [hunk 1]
```diff
@@ -112,6 +113,26 @@ For example, to use the :code:`ubuntu:20.04` image from Docker Hub:
...
+
+  setup: |
+    # Commands to run inside the container
+
+  run: |
+    # Commands to run inside the container
+
+    # Since SkyPilot tasks are run inside a fresh conda "(base)" environment,
+    # deactivate first to access what the Docker image has already installed.
+    source deactivate
```
Comment (Michaelvll) [inline docs/source/examples/docker-containers.rst:132]: This should be fixed by #2801

File: docs/source/reference/faq.rst [hunk 2]
```diff
@@ -93,6 +93,41 @@ To launch a VS Code tunnel using a SkyPilot task definition, you can use the fol
...
+      - region: europe-west1
+      - region: europe-west10
+      - region: europe-west12
+      - region: europe-west2
+      - region: europe-west3
+      - region: europe-west4
+      - region: europe-west6
+      - region: europe-west8
+      - region: europe-west9
+      # Or put in other clouds' Europe regions.
```
Comment (Michaelvll) [inline docs/source/reference/faq.rst:127]: In the future, we may want to support some predefined areas, such as
```
resources:
  region: gdpr-area
```

Also, this task yaml will make it quite complicated to also select some accelerators, as we have to add the same accelerator for all the items in `any_of`. In the future, we should be able to take default fields and additional `any_of`, i.e.:
```yaml
resources:
  accelerators: V100
  any_of:
      # AWS:
      - region: eu-central-1
      - region: eu-west-1
      - region: eu-west-2
      - region: eu-west-3
      - region: eu-north-1
      ...
```

**Review by Michaelvll — APPROVED** [review]
Thanks for the update @concretevitamin! The changes LGTM. I listed several TODOs in the comment.

File: docs/source/reference/faq.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/reference/faq.rst:127]: Yep, taking defaults + treating `any_of` and `ordered` as overrides sounds good.

**Pull request merged**